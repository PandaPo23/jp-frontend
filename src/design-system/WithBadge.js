import PropTypes from 'prop-types';
import React from 'react';

import Absolute from './Absolute';
import Badge from './Badge';
import Relative from './Relative';

const WithBadge = ({
  children,
  label,
  position = { top: '.5em', left: 0 },
  bg,
  color,
  ...props
}) => (
  <Relative {...props}>
    {children}
    <Absolute {...position}>
      <Badge error>{label}</Badge>
    </Absolute>
  </Relative>
);

WithBadge.displayName = 'WithBadge';

WithBadge.propTypes = {
  content: PropTypes.string || PropTypes.number,
  position: PropTypes.string,
  bg: PropTypes.string,
  color: PropTypes.string,
};

export default WithBadge;
