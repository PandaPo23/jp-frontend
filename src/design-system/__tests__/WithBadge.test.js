import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import WithBadge from '../WithBadge';

describe('WithBadge component', () => {
  it('renders content with badge correctly', () => {
    const localProps = {
      bagdeLabel: '3',
      position: 'right',
      children: <div>Content</div>,
    };
    const wrapper = shallow(<WithBadge {...localProps} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
