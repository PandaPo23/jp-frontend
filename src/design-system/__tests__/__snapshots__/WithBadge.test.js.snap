// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WithBadge component renders content with badge correctly 1`] = `
<Relative
  bagdeLabel="3"
  position="relative"
>
  <div>
    Content
  </div>
  <Absolute
    0="r"
    1="i"
    2="g"
    3="h"
    4="t"
    position="absolute"
  >
    <Badge
      borderRadius="circle"
      error={true}
      fontSize={3}
      letterSpacing="caps"
      opacity={0.8}
      px={1}
      textAlign="center"
      textTransform="uppercase"
    />
  </Absolute>
</Relative>
`;
